lib/get-exports.js

```diff
@@ -27,6 +27,72 @@ function getExportsForNodeBuiltIn (name) {
 
 const urlsBeingProcessed = new Set() // Guard against circular imports.
 
+/**
+ * This function looks for the package.json which contains the specifier trying to resolve.
+ * Once the package.json file has been found, we extract the file path from the specifier
+ * @param {string} specifier The specifier that is being search for inside the imports object
+ * @param {URL|string} fromUrl The url from which the search starts from
+ * @returns file to url to file export
+ */
+function resolvePackageImports (specifier, fromUrl) {
+  if (!specifier.startsWith('#')) {
+    return null
+  }
+
+  try {
+    const fromPath = fileURLToPath(fromUrl)
+    let currentDir = dirname(fromPath)
+
+    // search for package.json file which has the real url to export
+    while (currentDir !== dirname(currentDir)) {
+      const packageJsonPath = join(currentDir, 'package.json')
+
+      let packageJson
+      try {
+        packageJson = JSON.parse(readFileSync(packageJsonPath, 'utf8'))
+      } catch (error) {
+        if (error.code === 'ENOENT') {
+          currentDir = dirname(currentDir)
+          continue
+        }
+        throw error
+      }
+
+      if (packageJson) {
+        if (packageJson.imports && packageJson.imports[specifier]) {
+          const imports = packageJson.imports[specifier]
+
+          // Look for path inside packageJson
+          let resolvedPath
+          if (Object.prototype.toString.call(imports) === '[object Object]') {
+            const requireSpecifier = imports.require
+            const importSpecifier = imports.import
+            // look for the possibility of require and import which is standard for CJS/ESM
+            if (requireSpecifier || importSpecifier) {
+              // trying to resolve based on order of importance
+              resolvedPath = requireSpecifier.node || requireSpecifier.default || importSpecifier.node || importSpecifier.default
+            } else if (imports.node || imports.default) {
+              resolvedPath = imports.node || imports.default
+            }
+          } else if (typeof imports === 'string') {
+            resolvedPath = imports
+          }
+
+          if (resolvedPath) {
+            return pathToFileURL(require.resolve(resolvedPath, { paths: [currentDir] })).href
+          }
+        }
+        // return if we find a package.json but did not find an import
+        return null
+      }
+    }
+  } catch (error) {
```

Comment:
```suggestion
  } catch {
```

Alternatively, we could also just add the `cause`